shremote.py

```diff
@@ -524,60 +543,41 @@ def __init__(self, program_name, cmd_cfg, index = None):
 
         if 'begin' in cmd_cfg:
             begin_raw = cmd_cfg.formatted('begin')
-            try:
-                begin_evaled = safe_eval(begin_raw, 'begin')
-                if str(begin_evaled) != str(begin_raw): # Means that 'eval' did something
-                    log_warn("Specifying evaluatable begin without $(...) is deprecated: {}".format(begin_raw))
-                self.begin = float(begin_evaled)
-            except TypeError: # Begin shouldn't have to be eval'd again
-                self.begin = float(begin_raw)
+            self.begin = try_eval(begin_raw, 'begin')
 
         self.index = index if index is not None else 0
         self.program = Program.get(program_name)
-
         self.log = Log.get(program_name)
         self.name = program_name
 
+        if self.program.start is None:
+            log_fatal("{}: Program requires a 'start' command".format(program_name))
```

Comment:
I had used a stop-only program a few times in the past to ensure that programs are killed at the end of execution. Something like a `stop_all` program.

I'm not using it anymore so I'm okay with mandating that program.start exists. Just excusing my own messy behavior.